publish_courses/api.py

```diff
@@ -152,178 +130,119 @@ def create(self, request, *args, **kwargs) -> Response:
+        a list of course IDs and their job detail IDs (or record id from the db).
+        
+        Payloads are divided into chunks when to avoid exceeding the payload size limit for Lambda 
+        invocations. Each chunk is sent to the SQS queuing Lambda as an event. This is a fire and 
+        forget method and will not wait for the Lambda to finish.
+        """
+        logger.info(f'Sending bulk publish courses job ID {job.id} to queuing Lambda {QUEUEING_LAMBDA_NAME}.')
+        
+        try:
+            # Create Lambda client
+            lambda_client = boto3.client('lambda')
+        except ClientError as e:
+            logger.error(f'Error configuring Lambda client: {e}.', exc_info=True)
+            raise
+        except Exception as e:
+            logger.exception('Error configuring Lambda client.')
+            raise
 
-        logger.info(
-            f'The bulk publish courses job creation is complete and has been sent to the SQS queue.')
-        return Response(status=status.HTTP_201_CREATED)
+        max_batch_size = 2500  # To avoid exceeding the payload size limit for Lambda invocations. https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html
+        chunks = [job_details_list[x:x + max_batch_size] for x in range(0, len(job_details_list), max_batch_size)]
+        for chunk in chunks:
+            # Construct the job dictionary (will be used by `send_job_to_queueing_lambda` func as payload).
+            job_dict = {
+                "job_id": job.id,
+                "course_list": [{"course_id": jo.canvas_course_id, "job_detail_id": jo.id} for jo in chunk]
+            }
+
+            # Invoke Lambda function.
+            response = lambda_client.invoke(
+                FunctionName=QUEUEING_LAMBDA_NAME,
+                InvocationType='Event',
+                Payload=json.dumps(job_dict)
+            )
+
+            logger.info(f'Sent bulk publish courses job ID {job.id} to queuing Lambda {QUEUEING_LAMBDA_NAME}.',
+                        extra={'response': response})
+            
+        return None
 
 
-    def get_canvas_courses(self, account: str, term: str) -> List[Dict]:
+def _get_courses(account: str, term: str) -> List:
         """
-        Fetches a list of Canvas courses for a specified account and term. 
+        Returns a list of canvas courses for the given account and term.
         """
-        logger.debug(f'Retrieving Canvas courses.', extra={
-                     'account': account, 'term': term})
+        logger.info(f"Retrieving all courses for {account} and term {term}.")
 
-        canvas_courses = []
         try:
             canvas_courses = get_all_list_data(
                 SDK_CONTEXT,
                 list_active_courses_in_account,
                 account_id=account,
-                enrollment_term_id=term,
-                published=False,
+                enrollment_term_id=term
```

Comment:
You can pass the published=False parameter here to only retrieve unpublished courses btw. It would eliminate the additional loop for filtering